### PR TITLE
Add GitHub Action to build and deploy

### PR DESCRIPTION
**Pull request by Fryguy**

### CODE REVIEW

**Comment by Fryguy** [comment]
@bdunne This is ready to go.  I've tested most of the functionality on the setup side for the ibmcloud cli/kubectl via GHA on my fork.  The only thing I didn't test was the docker build/push stuff from within GHA.  That docker build/push code always worked locally, and the Kube deployment change code also always worked locally  (though I did refactor both in this PR).

File: .github/workflows/build_and_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        ruby-version: "2.7"
+        bundler-cache: true
+    - name: Docker login
+      run: echo ${{ secrets.DOCKER_REGISTRY_PASSWORD }} | docker login docker.io --password-stdin --username ${{ secrets.DOCKER_REGISTRY_USERNAME }}
+    - name: Build
+      run: bundle exec rake production:release:build[${{ github.ref_name }}]
+    - name: Set up kubectl
+      uses: azure/setup-kubectl@v4
+    - name: Set up ibmcloud CLI
+      uses: IBM/actions-ibmcloud-cli@4fb9b6c
```
Comment (Fryguy) [inline .github/workflows/build_and_deploy.yml:24]: TODO: This is in progress - I'm working to revamp this action - Eventually when I cut a release I'll switch this to `@v1`

File: lib/tasks/production.rake [hunk 2]
```diff
@@ -131,56 +140,75 @@ namespace :production do
...
+    desc "Build the specified version and push to docker.io"
+    task :build, [:version] do |_t, args|
+      version = release_version(args[:version])
+      image   = "docker.io/manageiq/miq_bot:#{version}"
+
+      puts "Building docker image..."
+      exit 1 unless system("docker build . --no-cache --build-arg REF=#{version} -t #{image}")
+      puts
+      puts "Pushing docker image..."
+      exit 1 unless system("docker login") && system("docker push #{image}")
```
Comment (bdunne) [inline lib/tasks/production.rake:189]:
```suggestion
      exit 1 unless system("docker login docker.io") && system("docker push #{image}")
```

File: .github/workflows/build_and_deploy.yml (hunk 1 above)
Comment (Fryguy) [reply .github/workflows/build_and_deploy.yml]: @bdunne I released v1, so I've updated this accordingly

**Comment by Fryguy** [comment]
@bdunne There's some package updates I'm going to merge after this is done, and then use that to test the deployment works.

**Pull request merged**